src/main/webapp/script.js

```diff
@@ -13,7 +13,58 @@
 // limitations under the License.
 
 function addThreads() {  // eslint-disable-line
-  fetch('/data').then((response) => response.text()).then((quote) => {
-    document.getElementById('thread-container').innerText = quote;
+  const url = '/data';
+  fetch(url).then((response) => response.json()).then((threadInfoList) => {
+    const threadList = document.getElementById('thread-container');
+    threadList.innerHTML = '';
+
+    let i;
+    for (i = 0; i < threadInfoList.length; i += 4) {
```

Comment:
Add TODO comment to change to parsing an object.

Reply:
I was thinking of changing this code so a json object contains a key for each aspect of the description (eg. title, sentiment etc.)  which would have the associated values for that specific key.